apps/backend/src/modules/user-follows/user-follows.repository.ts

```diff
@@ -0,0 +1,44 @@
+import { UserFollowsModel } from "~/modules/user-follows/user-follows.model.js";
+
+class UserFollowsRepository {
+	private userFollowsModel: typeof UserFollowsModel;
+
+	public constructor(userFollowsModel: typeof UserFollowsModel) {
+		this.userFollowsModel = userFollowsModel;
+	}
+
+	public async checkIsUserFollowing(
+		followerId: number,
+		followingId: number,
+	): Promise<boolean> {
+		const relation = await UserFollowsModel.query()
+			.where({ followerId, followingId })
+			.first();
+
+		return Boolean(relation);
+	}
+
+	public async followUser(
+		followerId: number,
+		followingId: number,
+	): Promise<void> {
+		await UserFollowsModel.query().insert({
+			followerId,
+			followingId,
+		} as Partial<UserFollowsModel>);
```

Comment:
do we need to cast it at all? I see we don't use casting in other places
```suggestion
		});
```

Reply:
TS2589: Type instantiation is excessively deep and possibly infinite.

i added this because i got an error